src/main/java/com/arcbees/bourseje/client/admin/dashboard/AdminDashboardView.java

```diff
@@ -114,18 +94,16 @@ public void f() {
     }
 
     @Override
-    public void setNumberOfVotesForCandidate(CandidateResult candidateResult) {
-        SpanElement element = numberOfVoteElements.get(candidateResult.getCandidateName());
-
-        if (element == null) {
-            GQuery.console.error("Candidate not found: " + candidateResult.getCandidateName());
-        } else {
-            element.setInnerText(String.valueOf(candidateResult.getNumberOfVotes()));
-        }
+    public void setCurrentState(VoteState currentState) {
+        this.currentState.setInnerText(currentState.toString());
```

Comment:
You should use a renderer for VoteState. toString for display purposes is not good.